src/PrestaShopBundle/Controller/Admin/Improve/Modules/ModuleAbstractController.php

```diff
@@ -89,6 +91,42 @@ protected function getToolbarButtons()
             ];
         }
 
-        return $toolbarButtons;
+        return array_merge($toolbarButtons, $this->getExtraToolbarButtons());
+    }
+
+    /**
+     * This method will call the actionAdminModuleExtraToolbarButton hook and allow
+     * modules to add some extra buttons to the modules catalogue toolbar.
+     * All the button's specification keys ('href', 'desc', 'icon', 'help') are mandatory
+     */
+    private function getExtraToolbarButtons(): array
+    {
+        try {
+            $extraToolbarContentFromHooks = (new HookFinder())
+                ->setHookName('actionAdminModuleExtraToolbarButton')
+                ->setParams(['controller' => $this])
+                ->find();
+        } catch (CoreException $exception) {
+            return [];
+        }
+
+        $extraToolbarButtons = [];
+
+        // Validation. We check that we have the exact keys
+        foreach ($extraToolbarContentFromHooks as $moduleName => $extraToolbarContentFromHook) {
+            if (!is_array($extraToolbarContentFromHook)) {
+                continue;
+            }
+
+            foreach ($extraToolbarContentFromHook as $buttonIndex => $extraToolbarButton) {
+                if (!empty(array_diff(['href', 'desc', 'icon', 'help'], array_keys($extraToolbarButton)))) {
```

Comment:
specification is strict here but looking at `admin_dev/themes/default/template/page_header_toolbar.tpl`, only href and desc are mandatory